OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -714,27 +714,25 @@ internal static class DreamProcNativeRoot {
         [DreamProcParameter("Needle", Type = DreamValueTypeFlag.String)]
         [DreamProcParameter("Start", Type = DreamValueTypeFlag.Float, DefaultValue = 1)]
         [DreamProcParameter("End", Type = DreamValueTypeFlag.Float, DefaultValue = 0)]
-        public static DreamValue NativeProc_findtext(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
+        public static async Task<DreamValue> NativeProc_findtext(AsyncNativeProc.State state) {
             // TODO This is for handling nulls, check if it works right for other bad types
+            DreamValue haystack = state.GetArgument(0, "Haystack");
+            int start = state.GetArgument(2, "Start").MustGetValueAsInteger(); //1-indexed
+            int end = state.GetArgument(3, "End").MustGetValueAsInteger(); //1-indexed
             int failCount = 0;
-            if (!bundle.GetArgument(0, "Haystack").TryGetValueAsString(out var text)) {
-                failCount++;
-            }
 
-            DreamValue needleArg = bundle.GetArgument(1, "Needle");
-            DreamObjectRegex? regex = null;
-            if (!needleArg.TryGetValueAsString(out var needle)) {
-                if(!needleArg.TryGetValueAsDreamObject(out regex)) {
-                    failCount++;
-                }
+            if (!haystack.TryGetValueAsString(out var text)) {
+                failCount++;
             }
 
-            if (failCount > 0) {
-                return new DreamValue(failCount == 2 ? 1 : 0);
+            DreamValue needleArg = state.GetArgument(1, "Needle");
+            if (needleArg.TryGetValueAsDreamObject<DreamObjectRegex>(out var regexObject)) {
+                // According to the docs, this is the same as /regex.Replace()
+                return await DreamProcNativeRegex.RegexFind(state, regexObject, haystack, start, end);
             }
 
-            int start = bundle.GetArgument(2, "Start").MustGetValueAsInteger(); //1-indexed
-            int end = bundle.GetArgument(3, "End").MustGetValueAsInteger(); //1-indexed
+            if(!needleArg.TryGetValueAsString(out var needle))
+                return DreamValue.True;
```

Comment:
BYOND returns different values for different combinations of invalid arguments. Are those still replicated?